app/src/main/java/org/mozilla/focus/session/SessionNotificationService.kt

```diff
@@ -150,17 +151,24 @@ class SessionNotificationService : Service() {
             val intent = Intent(context, SessionNotificationService::class.java)
             intent.action = ACTION_START
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                context.startForegroundService(intent)
-            } else {
-                context.startService(intent)
-            }
+            // Post starting the service to the end of the main thread to give it the best
+            // chance to complete in time.
+            ThreadUtils.postToMainThread(Runnable {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    context.startForegroundService(intent)
+                } else {
+                    context.startService(intent)
+                }
+            })
         }
 
         internal fun stop(context: Context) {
             val intent = Intent(context, SessionNotificationService::class.java)
 
-            context.stopService(intent)
+            // Since we're posting start, we should also post stop
```

Comment:
nit: You should explain why that is: to ensure they're run in the order they were called in, you have to put them in the same sequential queue.